### ISSUE TITLE
screen.close() + Raspian?

### ISSUE THREAD

**Issue by EweLoHD** [mabe02/lanterna#356]
Hey, I'm have written some Code with laterna on my Windows Pc and it worked fine, but when I ran my Code on my RaspberryPi and I press the <Enter> Button, so It should close the Screen, my code freezes and does nothing happens. The <Cancel> Button works. I'm using Java 8 and as SSH Client I'm using Terminus (I tried also Putty).

### Screenshot:
![https://i.imgur.com/WuKuVhi.png](https://i.imgur.com/WuKuVhi.png)

### My Code:
```java
public String token(){
	String t =null;
	try {
	   input = new FileInputStream("config.properties");
	   prop.load(input);

            //Terminal
            Terminal terminal = new DefaultTerminalFactory().createTerminal();
            Screen screen = new TerminalScreen(terminal);
            screen.startScreen();

            //ComboBox
            ComboBox<String> comboBox = new ComboBox<String>();
            comboBox.addItem("Normal");
            comboBox.addItem("Dev");

            //Panel
            Panel panel = new Panel();
            panel.setLayoutManager(new GridLayout(3));

            //Label
            panel.addComponent(new Label("Choose a Bot: "));

            //Add Combobox
            panel.addComponent(comboBox);

            //Add some Space
            panel.addComponent(new EmptySpace(new TerminalSize(3,1)));
            panel.addComponent(new EmptySpace(new TerminalSize(3,1)));
            panel.addComponent(new EmptySpace(new TerminalSize(3,1)));
            panel.addComponent(new EmptySpace(new TerminalSize(3,1)));
            panel.addComponent(new EmptySpace(new TerminalSize(3,1)));

            //Add Button
            Button button = new Button("Enter", new Runnable() {

                @Override
                public void run() {
                    try {
                        System.out.println("hi");
                        screen.close();
                    } catch (IOException e) {
                        e.printStackTrace();
                    }
                }
            });
            panel.addComponent(button);

            Button buttonc = new Button("Cancel", new Runnable() {

                @Override
                public void run() {
                    System.exit(1);
                }
            });
            panel.addComponent(buttonc);


            // Create window to hold the panel
            BasicWindow window = new BasicWindow();
            window.setHints(Arrays.asList(Window.Hint.CENTERED));
            window.setComponent(panel);

            // Create gui and start gui
            MultiWindowTextGUI gui = new MultiWindowTextGUI(screen, new DefaultWindowManager(), new EmptySpace(TextColor.ANSI.BLUE));
            gui.addWindowAndWait(window);


            if(comboBox.getSelectedIndex() == 0){
                t = prop.getProperty("token");
            }
            else {
                t = prop.getProperty("token_dev");
            }
            t = prop.getProperty("token");
		} catch (IOException e) {
			e.printStackTrace();
		}
		return t;
	}
```

**Comment by avl42** [comment]
I think, the preferred way to end a textgui app is to close the main window.

on windows, where you likely had a swingterminal, the swingterminal
indirectly reacts to closing screen, because closing the screen includes
exiting the terminal's "private" mode, and swingterminal takes this as hint
to close itself, (unless told otherwise).

on raspian, closing the screen leaves the terminal running, and apparently
also the textgui, despite not really able to draw itself on a stopped
screen.

long story short: rather close the window than stop the screen to end the
app.

i'm also a bit surprised by that "white artifact" near the bot selection on
your screenshot, but i don't have my devel machine at hand right now to
examine that further...

>

**Comment by EweLoHD** [comment]
Thank you very much! It (nearly) works now, but there is an other Problem: When the Window is closed and it prints something out it looks very buggy and I can't type something in:

![https://i.imgur.com/o0ZQ8uR.png](https://i.imgur.com/o0ZQ8uR.png)

The White Artifact might have something to do with Terminus and its Theme.

**Comment by avl42** [comment]
seems like your Java code complains about a broken (or missing) "slf4j"-configuration, or some missing slf4j jar files, so hence those errors (or are they warnings?)  visiting that link in those messages might help getting rid of these messages.

Anyway, either due to the exception or maybe the System.exit(1) call on cancel, your terminal might still be in raw&noecho mode, which makes it appear as if it were impossible to type. Usually in that case, there would still be a shell prompt written out, but I can't see any on your screen shot.

see if blindly entering commands and pressing return key causes any reaction, if so, try command "stty sane", which typically resets the terminal settings to sane defaults, after which you once again see your typing.

Summary: System.exit() may leave the terminal in a bad state.   typing "stty sane" usually clears that if after exiting the Java program you fall back to a unix shell (like bash).  if your java app got called from another program (or script), you might add a call to "stty sane" from there, just in case the app crashes. Normally, lanterna apps successfully reset the terminal themselves.

**Comment by EweLoHD** [comment]
Ok thank you very much.
